corsheaders/validators.py

```diff
@@ -0,0 +1,10 @@
+def return_type(*require_type):
+    def check_type(func):
+        def decorator_wrapper(*args, **kwargs):
+            return_value = func(*args, **kwargs)
+            return_value_type = type(return_value)
+            if not return_value_type in require_type:
```

Comment:
In Python should always do `isinstance(x, desired_type)` to check types, not `type(x) == desired_type`, to allow subclassing. Also even better is to use the abstract types in `collections.abc`, for example `Sequence`!